contracts/UniversalV3Staker.sol

```diff
@@ -273,23 +273,12 @@ contract UniversalV3Staker is IUniversalV3Staker, Multicall {
         deposits[tokenId].numberOfStakes--;
         incentive.numberOfStakes--;
 
-        (, uint160 secondsPerLiquidityInsideX128, ) =
-            key.pool.snapshotCumulativesInside(deposit.tickLower, deposit.tickUpper);
-        (uint256 reward, uint160 secondsInsideX128) =
-            RewardMath.computeRewardAmount(
-                incentive.totalRewardUnclaimed,
-                incentive.totalSecondsClaimedX128,
-                key.startTime,
-                key.endTime,
-                liquidity,
-                secondsPerLiquidityInsideInitialX128,
-                secondsPerLiquidityInsideX128,
-                block.timestamp
-            );
+        uint24 tickLowerShifted = uint24(deposit.tickLower - TickMath.MIN_TICK + 1);
+        uint24 tickUpperShifted = uint24(deposit.tickUpper - TickMath.MIN_TICK + 1);
+        uint256 latestReward = _calculateReward(liquidity, tickLowerShifted, tickUpperShifted);
+        uint256 reward = latestReward - rewardDebt;
+        require(reward <= latestReward, 'UniswapV3Staker::getRewardInfo: overflow');
 
-        // if this overflows, e.g. after 2^32-1 full liquidity seconds have been claimed,
-        // reward rate will fall drastically so it's safe
-        incentive.totalSecondsClaimedX128 += secondsInsideX128;
         // reward is never greater than total reward unclaimed
         incentive.totalRewardUnclaimed -= reward;
```

Comment:
I think totalRewardUnclaimed could be deleted.

Reply:
left here for book keeping reasons: `updatePrice` will check if the incentive exists or not and this field can be used.  we may clean up all unused variables in a later PR